### PR TITLE
chore: update Best

### PR DESCRIPTION
**Pull request by sf-v**
## Details

Update Best to the latest version - Best now uses the latest version of Rollup!

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -2,9 +2,7 @@ name: Run Best benchmark
 
 # We could run on every PR, but there is not much need right now.
 # By running on every push to master, we get historical Best data for the master branch.
-on:
-  push:
-    branches: [ master ]
+on: push
```
Comment (nolanlawson) [inline .github/workflows/benchmark.yml:5]: I assume this is just a temporary change to test, right?

File: packages/@lwc/perf-benchmarks/package.json [hunk 2]
```diff
@@ -23,10 +23,10 @@
...
     },
     "devDependencies": {
-        "@best/cli": "^9.0.1",
-        "@best/runner-headless": "^9.0.1",
-        "@best/runner-remote": "^9.0.1",
-        "@lwc/rollup-plugin-node-resolve-legacy": "npm:rollup-plugin-node-resolve@5.2.0",
+        "@best/cli": "^10.0.0-alpha.0",
+        "@best/runner-headless": "^10.0.0-alpha.0",
+        "@best/runner-remote": "^10.0.0-alpha.0",
+        "@rollup/plugin-node-resolve": "15.2.3",
```
Comment (nolanlawson) [inline packages/@lwc/perf-benchmarks/package.json:29]:
```suggestion
```

This is already defined in the top-level `package.json`. No need to re-define it here.

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (sf-v) [reply .github/workflows/benchmark.yml:5]: @nolanlawson Yes, just testing things to make sure everything is working.

File: packages/@lwc/perf-benchmarks/package.json (hunk 2 above)
Comment (sf-v) [reply packages/@lwc/perf-benchmarks/package.json]: @nolanlawson Done!

File: packages/@lwc/perf-benchmarks/package.json [hunk 3]
```diff
@@ -23,10 +23,9 @@
         "@lwc/synthetic-shadow": "5.0.0"
     },
     "devDependencies": {
-        "@best/cli": "^9.0.1",
-        "@best/runner-headless": "^9.0.1",
-        "@best/runner-remote": "^9.0.1",
-        "@lwc/rollup-plugin-node-resolve-legacy": "npm:rollup-plugin-node-resolve@5.2.0",
+        "@best/cli": "^10.0.0-alpha.0",
+        "@best/runner-headless": "^10.0.0-alpha.0",
+        "@best/runner-remote": "^10.0.0-alpha.0",
```
Comment (sf-v) [inline packages/@lwc/perf-benchmarks/package.json:28]: @nolanlawson Note: I'll update this PR once v10.0.0 is released (currently checking to see how different projects handle this change).

**Comment by sf-v** [comment]
@nolanlawson The PR is now ready!

**Pull request merged**